src/chemistry/geoschem/cesmgc_history_mod.F90

```diff
@@ -0,0 +1,1191 @@
+             EXIT
+          ENDIF
+         ELSEIF ( current%vloc == VLocationEdge ) THEN
+            CALL addfld(trim(current%name),                                 &
+                        (/'ilev'/),                                         &
+                        'I',                                                &
+                        trim(current%units),                                &
+                        trim(current%long_name)                            )
+         ELSE
+            IF ( am_I_Root ) THEN
+               PRINT *, "Unknown vertical location for ", &
+                        TRIM(current%name)
+            ENDIF
+         ENDIF
+       ELSEIF ( current%rank == 2 ) THEN
+          CALL addfld(trim(current%name),                                 &
+                      horiz_only,                                         &
+                      'I',                                                &
+                      trim(current%units),                                &
+                      trim(current%long_name)                            )
+          IF ( RC == GC_FAILURE ) THEN
+             ErrMsg =  "Problem adding 2D export for " // TRIM(current%name)
+             EXIT
+          ENDIF
+       ELSE
+          RC = GC_FAILURE
+          ErrMsg = "Problem adding export for " // TRIM(current%name) // &
+                   ". Rank is only implemented for 2 or 3!"
+          EXIT
+       ENDIF
+
+       current => current%next
+    ENDDO
+    current => NULL()
+
+    IF ( RC == GC_FAILURE ) THEN
+       CALL GC_ERROR( ErrMsg, RC, Iam )
+       _ASSERT(.FALSE., 'informative message here')
+       RETURN
+    ENDIF
+
+  END SUBROUTINE HistoryExports_SetServices
+!EOC
+!------------------------------------------------------------------------------
+!                  GEOS-Chem Global Chemical Transport Model                  !
+!------------------------------------------------------------------------------
+!BOP
+!
+! !IROUTINE: CopyGCStates2Exports
+!
+! !DESCRIPTION:
+!\\
+!\\
+! !INTERFACE:
+!
+  SUBROUTINE CopyGCStates2Exports( am_I_Root, Input_Opt, State_Grid, HistoryConfig, LCHNK, RC )
+!
+! !USES:
+!
+  USE HCO_Interface_GC_Mod, ONLY : HCOI_GC_WriteDiagn
+  USE Input_Opt_Mod,    ONLY : OptInput
+  USE State_Grid_Mod,   ONLY : GrdState
+
+  USE cam_history,      ONLY : hist_fld_active, outfld
+  USE SHR_KIND_MOD,     ONLY : shr_kind_r8
+!
+! !INPUT PARAMETERS:
+!
+    LOGICAL,        INTENT(IN)    :: am_I_Root
+    TYPE(OptInput), INTENT(IN)    :: Input_Opt
+    TYPE(GrdState), INTENT(IN)    :: State_Grid
+    INTEGER,        INTENT(IN)    :: LCHNK            ! Chunk number for CESM
+!
+! !INPUT AND OUTPUT PARAMETERS:
+!
+    TYPE(HistoryConfigObj), POINTER :: HistoryConfig  ! History config object
+!
+! !OUTPUT PARAMETERS:
+!
+    INTEGER,            INTENT(OUT) :: RC
+!
+! !REMARKS:
+!  !
+! !REVISION HISTORY:
+!  01 Sep 2017 - E. Lundgren - Initial version
```

Comment:
I think it would make sense to edit the revision histories in this file to reflect you adapted this from my work. What do you think of keeping initial version with my name, but putting "Initial version for GCHP/GEOS". Then adding another line with your name, date, and "Adapted for CESM" for subroutines you changed?

Reply:
All done. Thanks for the reminder.